File: Andriod/Practical/OasisSnack/app/src/main/java/com/oasissnacks/oasissnacks/adapter/VIewPagerAdapter.java
package com.oasissnacks.oasissnacks.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.oasissnacks.oasissnacks.fragment.DecriptionFragment;
import com.oasissnacks.oasissnacks.fragment.DisclamierFragment;
import com.oasissnacks.oasissnacks.fragment.MoreInfoFragment;

public class VIewPagerAdapter extends FragmentPagerAdapter {
    public VIewPagerAdapter(@NonNull FragmentManager fm) {
        super(fm);
    }


    @Override
    public Fragment getItem(int position) {
        Fragment fragment = null;
        switch (position) {
            case 0:
                fragment = new DecriptionFragment();
                break;
            case 1:
                fragment = new DisclamierFragment();
                break;
            case 2:
                fragment = new MoreInfoFragment();
                break;
                default:break;
        }

        return fragment;
    }

    @Override
    public int getCount() {
        return 3;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        String title = null;
        if (position == 0) {
            title = "Description";
        } else if (position == 1) {
            title = "Disclaimer";
        } else if (position == 2) {
            title = "More Information";
        }
        return title;
    }
}

File: Andriod/Practical/usl_kotlin/app/src/main/java/com/usl/usl/adapter/ViewPagerAdapter.kt
package com.usl.usl.adapter

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.usl.usl.fragment.DSectionFragment
import com.usl.usl.fragment.HSectionFragment
import com.usl.usl.fragment.MSectionFragment

class ViewPagerAdapter(fm: FragmentManager, var bundle: Bundle?) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        var fragment: Fragment? = null
        when (position) {
            0 -> fragment = MSectionFragment(bundle)
            1 -> fragment = DSectionFragment(bundle)
            2 -> fragment = HSectionFragment(bundle)
            else -> {
            }
        }
        return fragment!!
    }

    override fun getCount(): Int {
        return 3
    }

    override fun getPageTitle(position: Int): CharSequence? {
        var title: String? = null
        if (position == 0) {
            title = "M Section"
        } else if (position == 1) {
            title = "D Section"
        } else if (position == 2) {
            title = "H Section"
        }
        return title
    }


}